test/e2e/ps_test.go

```diff
@@ -334,25 +334,46 @@ var _ = Describe("Podman ps", func() {
 		Expect(result.OutputToString()).To(Equal(""))
 	})
 
-	It("podman ps id filter flag", func() {
-		_, ec, fullCid := podmanTest.RunLsContainer("")
+	It("podman ps ancestor filter with substring matching (Docker compatibility)", func() {
+		// Create a container to test with
+		_, ec, cid := podmanTest.RunLsContainer("test1")
 		Expect(ec).To(Equal(0))
 
-		result := podmanTest.Podman([]string{"ps", "-a", "--filter", fmt.Sprintf("id=%s", fullCid)})
-		result.WaitWithDefaultTimeout()
-		Expect(result).Should(ExitCleanly())
+		// Test with substring of image name (Docker compatibility)
+		result := podmanTest.PodmanExitCleanly("ps", "-q", "--no-trunc", "-a", "--filter", "ancestor=alpine")
+		Expect(result.OutputToString()).To(Equal(cid))
+
+		// Test with partial image name (should work with substring matching)
+		result = podmanTest.PodmanExitCleanly("ps", "-q", "--no-trunc", "-a", "--filter", "ancestor=alp")
+		Expect(result.OutputToString()).To(Equal(cid))
+
+		// Test with prefix substring of image ID (Docker compatibility, new functionality)
+		inspect := podmanTest.PodmanExitCleanly("inspect", cid, "--format", "{{.Image}}")
+		fullImageID := inspect.OutputToString()
+		imageIDPrefix := fullImageID[:12]
+		result = podmanTest.PodmanExitCleanly("ps", "-q", "--no-trunc", "-a", "--filter", "ancestor="+imageIDPrefix)
+		Expect(result.OutputToString()).To(Equal(cid))
+
+		// Test with non-prefix substring of image ID (Docker compatibility)
+		if len(fullImageID) > 16 {
+			imageIDSubstr := fullImageID[4:16]
+			result = podmanTest.PodmanExitCleanly("ps", "-q", "--no-trunc", "-a", "--filter", "ancestor="+imageIDSubstr)
+			Expect(result.OutputToString()).To(Equal(cid))
+		}
+
+		// Test with non-existent substring (should not match)
+		result = podmanTest.PodmanExitCleanly("ps", "-q", "--no-trunc", "-a", "--filter", "ancestor=nonexistent")
+		Expect(result.OutputToString()).To(Equal(""))
 	})
 
 	It("podman ps id filter flag", func() {
-		session := podmanTest.RunTopContainer("")
-		session.WaitWithDefaultTimeout()
-		Expect(session).Should(ExitCleanly())
-		fullCid := session.OutputToString()
+		_, ec, fullCid := podmanTest.RunLsContainer("")
+		Expect(ec).To(Equal(0))
 
-		result := podmanTest.Podman([]string{"ps", "-aq", "--no-trunc", "--filter", "status=running"})
+		result := podmanTest.Podman([]string{"ps", "-a", "--filter", fmt.Sprintf("id=%s", fullCid)})
 		result.WaitWithDefaultTimeout()
 		Expect(result).Should(ExitCleanly())
-		Expect(result.OutputToStringArray()[0]).To(Equal(fullCid))
+		Expect(result.OutputToStringArray()).To(ContainElement(ContainSubstring(fullCid)))
```

Comment:
and then if you use -q --no-trunc you can use
`Expect(result.OutputToString()).To(Equal(fullCid))` here again.